lib/active_merchant/billing/gateways/authorize_net.rb

```diff
@@ -268,11 +269,18 @@ def add_apple_pay_payment_token(xml, apple_pay_payment_token)
         end
       end
 
-      def add_retail_data(xml, payment)
-        return unless valid_track_data
-        xml.retail do
-          # As per http://www.authorize.net/support/CP_guide.pdf, '2' is for Retail, the only current market_type
-          xml.marketType(2)
+      def add_market_type(xml, payment)
+        return if card_brand(payment) == 'check' or card_brand(payment) == 'apple_pay'
+        if valid_track_data
+          xml.retail do
+            # marketType 2 = retail transaction
+            xml.marketType(2)
```

Comment:
Does it make sense to pull these magic numbers out into constants, so that you can remove the comment above?